scripts/gather_outdated_snapshots.py

```diff
@@ -0,0 +1,57 @@
+import datetime
+import json
+import os
+
+import click
+
+
+def get_outdated_snapshots_for_directory(path: str, date_limit: str) -> dict:
+    """
+    Fetches all snapshots that were recorded before the given date_limit
+    :param path: The directory where to look for snapshot files.
+    :param date_limit: All snapshots whose recorded-date is older than date-limit are considered outdated.
+            Format of the date-string must be "DD-MM-YYYY".
+    :return: List of test names whose snapshots (if any) are outdated.
+    """
+
+    date_limit = datetime.datetime.strptime(date_limit, "%d-%m-%Y")
+    result = {"date": date_limit}
+    outdated_snapshots = []
+    if not path.endswith("/"):
+        path = f"{path}/"
+    for file in os.listdir(path):
```

Comment:
we might need to check sub-folders as well, e.g. for s3 there are some snapshot files in a sub-directory